netqasm/sdk/compiling.py

```diff
@@ -266,8 +268,17 @@ def _move_carbon_electron(
     def _handle_two_qubit_gate(
         self, instr: core.TwoQubitInstruction
     ) -> List[NetQASMInstruction]:
-        qubit_id0 = self.get_reg_value(instr.reg0).value
-        qubit_id1 = self.get_reg_value(instr.reg1).value
+        try:
+            qubit_id0 = self.get_reg_value(instr.reg0).value
+            qubit_id1 = self.get_reg_value(instr.reg1).value
+        except KeyError:
+            # Register values are not known at compile time.
+            # We may assume that this is a MOV operation from the communication
+            # qubit to a memory qubit. (This is the only time a gate uses
+            # operands that are not known at compile time.)
+            assert isinstance(instr, vanilla.MovInstruction)
```

Comment:
shouldn't this assert not be done before the try?

Reply:
No, the instruction can be something other than a Mov. However, when there is no key found, we only allow the situation where it is a Mov.